main.jsonnet

```diff
@@ -0,0 +1,172 @@
+local dex = (import '../components/dex.libsonnet') + {
+  config+:: {
+    local cfg = self,
+    name: 'dex',
+    namespace: 'dex',
+    config+: {
+      oauth2: {
+        passwordConnector: 'local',
+      },
+      staticClients: [
+        {
+          id: 'test',
+          name: 'test',
+          secret: 'ZXhhbXBsZS1hcHAtc2VjcmV0',
+        },
+      ],
+      enablePasswordDB: true,
+      staticPasswords: [
+        {
+          email: 'admin@example.com',
+          // bcrypt hash of the string "password"
+          hash: '$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W',
+          username: 'admin',
+          userID: '08a8684b-db88-4b73-90a9-3cd1661f5466',
+        },
+      ],
+    },
+    version: 'v2.24.0',
+    image: 'quay.io/dexidp/dex:v2.24.0',
+    commonLabels+:: {
+      'app.kubernetes.io/instance': 'e2e-test',
+    },
+  },
+};
+
+local obs = (import '../environments/base/observatorium.jsonnet');
```

Comment:
and this?

Reply:
Yes, this needs to be fixed and we need to add a test case that checks if this compiles correctly.